### DISCUSSION TITLE
Our future for easy rollback!

### DISCUSSION THREAD

**Discussion by peaceiris** [pipe-cd/pipecd#4917] [Ideas]
I believe we need a place to discuss the rollback experience in PipeCD, which is why I opened this discussion.

Currently, to perform a rollback after a deployment is completed, it is necessary to create a new deployment by reverting the manifest or creating a pull request to revert the Docker image tag. This process seems a bit cumbersome. I am convinced that PipeCD has the potential to make this flow clearer and simpler.

**Comment by khanhtc1202** [comment]
Thanks, @peaceiris. I understand your point 👍. Do you currently have any ideal process or flow in mind? 
TBH, I think the current flow is a bit cumbersome, but it is clear. Please note that PipeCD is a GitOps tool, and it uses Git as its single source of trust, which means to PipeCD, each deployment should be a reflection of a manifest change commit. Happy to hear other opinions here cc @ffjlabo @t-kikuc

**Reply by peaceiris** [reply]
I think it would be convenient to be able to create a revert deployment triggered from a PipeCD application. For example, having a `Revert` button on the console page of a specific application, which, when clicked, makes the piped create a commit to revert the k8s manifest. Such a flow seems ideal.

**Reply by khanhtc1202** [reply]
It sounds nice 👍 I think this can be done easily since it looks similar to what we do for the EventWatch feature (I mean, we have a mechanism that allows piped to push/commit to the watching branch) 🤔

**Reply by peaceiris** [reply]
I agree with you. One concern I have is how PipeCD should manage the previous versions or commits.

**Reply by ffjlabo** [reply]
> I agree with you. One concern I have is how PipeCD should manage the previous versions or commits.

PipeCD Application has information on the latest successful deployment, including the commit.
https://github.com/pipe-cd/pipecd/blob/b1177abecc938d869804ad3eb068fa3d29821e6d/pkg/model/application.pb.go#L126-L127

So we might be able to create a revert commit and cancel the fix 🤔

**Reply by peaceiris** [reply]
That sounds good 😃

**Comment by peaceiris** [comment]
# Input rollback target Artifact Version

I mentioned the future of the `Revert` button in the above thread. In addition, we can discuss a roll backing to a specific version.

If we want to roll back to a specific Docker image tag, it might be useful to input any artifact version from the web console and create an automatic commit by the piped for the rollback.

<img width="404" alt="version" src="https://github.com/pipe-cd/pipecd/assets/30958501/e7d7182b-dd5b-4ad8-8243-a1f022a6f26b">
